### DISCUSSION TITLE
Global default configuration with plugins

### DISCUSSION THREAD

**Discussion by vincerubinetti** [prettier/prettier-vscode#3234] [Q&A]
I have been wasting hours of my life trying to figure this out, and fighting a host of huge unfixed bugs in this extension.

I work on many one-off, stand-alone `.js`/`.css`/`.svg`/etc. files that I want to format, and cannot and will not define a whole `.pretterrc`, `package.json`, and `node_modules` for each one.

So, what I need:

- A global default prettier configuration.
- No need for `package.json` or `node_modules` next to the file being formatted.
- Being able to import globally installed plugins, e.g. `prettier-plugin-css-order`.

Through much trial and error, I've been at least able to find a config that won't error and prevent Prettier from formatting:

**`~/.prettierrc.js`** (no other location or [filename](https://prettier.io/docs/en/configuration) would both be detected by the extension and not throw a config parsing error)

```js
module.exports = {
  plugins: [import("prettier-plugin-css-order")]
};
```

(That plugin is definitely installed globally with npm and yarn).

But, it doesn't show up as a plugin:

```
["INFO" - 3:29:01 PM] Prettier Options:
{
  "filepath": "/Users/vincerubinetti/Desktop/software-registry/test.css",
  "parser": "css",
  "plugins": [
    {}
  ]
}
```

I've also tried `require` (errors), and `~/../../../usr/local/lib/node_modules/prettier-plugin-css-order` and `/usr/local/lib/node_modules/prettier-plugin-css-order` and many other variations, and plain string versions of these. I restart VSCode after each change.

Can anyone advise?

**Comment by bjornde25** [comment]
turbo-rails (>= 0.9, < 2.0)
